### PR TITLE
Database upgrader

### PR DESCRIPTION
**Pull request by wxtim**
## Summary
Part of the work towards creating job platforms.

The database stores the host and batch_system, we would need to upgrade This could be done towards the end, you might need to put off upgrading the DB but instead upgrade the information coming out of the database.

### CODE REVIEW

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- cylc/flow/platform_lookup.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4769397&bid=15895842)

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- cylc/flow/platform_lookup.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4769397&bid=15895842)

File: cylc/flow/tests/test_rundb.py [hunk 1]
```diff
@@ -170,5 +196,76 @@ def test_upgrade_hold_swap():
         assert not dao.upgrade_is_held()
 
 
+def test_upgrade_to_platforms(tmp_path):
+    """Test upgrader logic for platforms in the database.
+    """
+    # Set up the globalrc
+    set_up_globalrc(GLOBALRC, tmp_path)
```
Comment (oliver-sanders) [inline cylc/flow/tests/test_rundb.py]: Where does `tmp_path` come from?

For bonus marks you could use a context manager something like this:

```python
with set_up_globalrc(GLOBALRC):
    # the temp file has been written
# the temp file has been removed
```

Comment (wxtim) [reply cylc/flow/tests/test_rundb.py]: `tmp_path` is a pytest fixture - it's just something that Pytest magically gives you see https://docs.pytest.org/en/latest/tmpdir.html

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- cylc/flow/platform_lookup.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4769397&bid=15895842)

**Comment by wxtim** [comment]
> I think old column should be harmless

Might need to be careful to avoid having these columns used in the near future lest we forget about them and end up with peculiar legacy issues later because that column is still in use somewhere else.

File: cylc/flow/tests/test_rundb.py (hunk 1 above)
Comment (oliver-sanders) [reply cylc/flow/tests/test_rundb.py]: Cool, learn something new every day. So long a pytest cleans up `tnp_path` afterwards that's great.

**Comment by oliver-sanders** [comment]
> column is still in use somewhere else

We should be bale to catch this with a quick `git grep`, shouldn't be too big a risk as the test battery should catch this too but good point. Perhaps add a TODO to the ticket to check for this before signing off on the job platforms work.

**Comment by oliver-sanders** [comment]
Codecov appears to be malfunctional reporting:

> Missing base report

File: cylc/flow/tests/test_rundb.py [hunk 2]
```diff
@@ -170,5 +196,76 @@ def test_upgrade_hold_swap():
...
+    # Set up the globalrc
+    set_up_globalrc(GLOBALRC)
+
+    # task name, cycle, user_at_host, batch_system
+    initial_data = [
+        ('hpc_with_pbs', '1', 'hpcl1', 'pbs'),
+        ('desktop_with_bg', '1', 'desktop01', 'background'),
+        ('slurm_no_host', '1', '', 'slurm'),
+        ('hpc_bg', '1', 'hpcl1', 'background'),
+        ('username_given', '1', 'slartibartfast@hpcl1', 'pbs')
```
Comment (oliver-sanders) [inline cylc/flow/tests/test_rundb.py]:
```suggestion
        ('username_given', '1', 'slartibartfast@deepthought', 'pbs')
```

Comment (wxtim) [reply cylc/flow/tests/test_rundb.py]: slartibartfast@the_earth, surely @oliver-sanders ?

Comment (oliver-sanders) [reply cylc/flow/tests/test_rundb.py]: slartibartfast@campain-for-real-time.mg

**Pull request merged**